packages/protocol/contracts/dapis/RrpDapiServer.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "../AirnodeRrp.sol";
+import "./reducers/MeanMedianHybrid.sol";
+
+contract RrpDapiServer is MeanMedianHybrid {
+    struct Dapi {
+        uint256 noResponsesToReduce;
+        int256 toleranceInPercentages;
+        uint256 requesterIndex;
+        bytes32[] templateIds;
+        address[] designatedWallets;
+        address reduceAddress;
+        bytes4 reduceFunctionId;
+        mapping(uint256 => int256[]) requestIndexToResponses;
+        mapping(uint256 => uint256) requestIndexToResponsesLength;
+        uint64 nextRequestIndex;
+        uint64 requestIndexResetCount;
+        address requestIndexResetter;
+        }
+    
+    struct DapiRequestIdentifiers {
+        bytes16 dapiId;
+        uint64 dapiRequestIndex;
+        uint64 dapiRequestIndexResetCountAtRequestTime;
+        }
```

Comment:
This was what I was referring to by optimizing storage by packing them into structs. The `Dapi` struct above has a lot of room for optimization for example.

Reply:
Yea, I was thinking of discussing this with you. Figuring out things like: max number of dAPIs, max number of requests, etc. In order to potentially change ints/uints to smaller types.

Reply:
Can be left to last imo